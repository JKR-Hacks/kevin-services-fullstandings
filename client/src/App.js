/* eslint-disable prefer-destructuring */
/* eslint-disable no-unused-vars */
import React, { Component } from 'react';
import axios from 'axios';

import FullStandings from './components/FullStandings';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      teams: [],
    };
  }

  componentDidMount() {
    axios.get('/espn/teamstandings',
      {
        method: 'GET',
        mode: 'no-cors',
        headers: {
          'Access-Control-Allow-Origin': '*',
          'Content-Type': 'application/json',
        },
      })
      .then((res) => {
        const data = res.data;
        this.setState({
          teams: data,
        });
      })
      .catch((err) => {
        console.log(err);
      });
  }

  render() {
    const { teams } = this.state;
    return (
      <div id="fullstandings">
        <FullStandings teams={teams} />
      </div>
    );
  }
}
