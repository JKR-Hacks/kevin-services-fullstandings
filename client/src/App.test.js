/* eslint-disable no-unused-vars */
/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import axios from 'axios';

import App from './App';

describe('App component', () => {
  it('Matches the snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render App component correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should update a state in a callback', () => {
    const wrapper = shallow(<App />);
    wrapper.setState({ scores: 'new' }, () => {
      expect(wrapper.state()).toEqual({ scores: 'new', teams: [], view: 'main' });
    });
  });

  it('Should call componentDidMount once', () => {
    const spy = jest.spyOn(App.prototype, 'componentDidMount');
    const wrapper = mount(<App />);
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
  });
});

describe('Make successful API call to database', () => {
  it('Should fetch via axios /espn/teamstandings', () => {
    const spy = jest.spyOn(App.prototype, 'componentDidMount');
    const wrapper = mount(<App />);
    wrapper.instance().componentDidMount(axios.get('/espn/teamstandings'));
    expect(spy).toHaveBeenCalled();
  });
});
