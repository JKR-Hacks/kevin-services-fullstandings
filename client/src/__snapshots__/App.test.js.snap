// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Matches the snapshot 1`] = `
<div
  id="fullstandings"
>
  <div
    className="container-fluid"
  >
    <h5>
      2018 NFL Standings
    </h5>
    <table
      className="table table-striped table-hover table-sm rounded"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Team
          </th>
          <th
            scope="col"
          >
            W
          </th>
          <th
            scope="col"
          >
            L
          </th>
          <th
            scope="col"
          >
            T
          </th>
          <th
            scope="col"
          >
            PCT
          </th>
          <th
            scope="col"
          >
            PF
          </th>
          <th
            scope="col"
          >
            PA
          </th>
          <th
            scope="col"
          >
            DIFF
          </th>
        </tr>
      </thead>
      <tbody>
        <h5>
          AFC EAST
        </h5>
      </tbody>
      <tbody>
        <h5>
          AFC NORTH
        </h5>
      </tbody>
      <tbody>
        <h5>
          AFC SOUTH
        </h5>
      </tbody>
      <tbody>
        <h5>
          AFC WEST
        </h5>
      </tbody>
      <tbody>
        <h5>
          NFC EAST
        </h5>
      </tbody>
      <tbody>
        <h5>
          NFC NORTH
        </h5>
      </tbody>
      <tbody>
        <h5>
          NFC SOUTH
        </h5>
      </tbody>
      <tbody>
        <h5>
          NFC WEST
        </h5>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`App component Should render App component correctly 1`] = `
<div
  id="fullstandings"
>
  <FullStandings
    teams={Array []}
  />
</div>
`;
